edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/identity/service/ServiceIdentity.cs

```diff
@@ -34,7 +34,7 @@ public class ServiceIdentity : IEquatable<ServiceIdentity>
             this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
             this.ModuleId = Option.Maybe(moduleId);
             this.DeviceScope = Option.Maybe(deviceScope);
-            this.ParentScopes = new List<string>(Preconditions.CheckNotNull(parentScopes, nameof(parentScopes)));
+            this.ParentScopes = parentScopes != null ? new List<string>(parentScopes) : new List<string>();
```

Comment:
We probably need to copy the DeviceScope property value to the ParentScopes array.

Reply:
@varunpuranik both `DeviceScopes` and `ParentScopes` were added in 1.2. So they both will be "empty" if we try to load 1.1 state. So, maybe things are a bit worse that just this issue.... what are the implications of empty scopes?

Reply:
I was under impression that if device is not authorized b/c of the (missing) scopes, we will still try to refresh the cache from upstream...

Reply:
yes, it needs to copy DeviceScope to ParentScopes list only if it's not an edge device. DeviceScopeIdentitiesCache assumes ParentScopes is set, it doesn't check DeviceScope and IotHub makes sure to sync the DeviceScope to ParentScopes for old leaf devices.

Reply:
@ancaantochi, what should happen if both `DeviceScope` and `ParentScopes` are empty (from old 1.1 state)?

Reply:
correct, if ParentScopes has value(s) it should just keep them, not to be overridden with DeviceScope.